### PR TITLE
55186 global search

### PR DESCRIPTION
**Pull request by jamesmk**
@tshedor Here's the global search feature so far. You can review it and merge it into v1.3 before taking it over on your branch.

### CODE REVIEW

File: app/assets/javascripts/fae/navigation/_navigation.js [hunk 1]
```diff
@@ -32,7 +30,7 @@ Fae.navigation = {
    */
   selectCurrentNavItem: function() {
     var current_base_url = window.location.pathname.replace('#', '');
-    var $ready_current_link = $('.js-nav a[href="' + current_base_url + '"]');
+    var $ready_current_link = $('.js-nav nav a[href="' + current_base_url + '"]');
```
Comment (tshedor) [inline app/assets/javascripts/fae/navigation/_navigation.js]: Why did you add this selector? This func adds active classes to both the sidenav and the header nav. Adding `nav` will narrow the scope to just the header nav.

File: app/assets/javascripts/fae/navigation/_navigation.js [hunk 2]
```diff
@@ -207,85 +205,6 @@ Fae.navigation = {
...
-        $search_wrapper.find('input').focus();
-      }
-
-    });
-  },
-
-  /**
-   * Initialize the typeahead navigation in the header nav
-   */
-  typeaheadSearch: function() {
```
Comment (tshedor) [inline app/assets/javascripts/fae/navigation/_navigation.js]: If you're scrubbing Typeahead, could you delete the files in `vendor/javascripts` too?

File: app/assets/javascripts/fae/navigation/_global_search.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    this.setElements();
+    this.utilitySearch();
+    this.searchListener();
+  },
+
+  /**
+   * Caches common elements into globally accessible variables
+   */
+  setElements: function() {
+    this.$wrapper = $('.utility-search-wrapper');
```
Comment (tshedor) [inline app/assets/javascripts/fae/navigation/_global_search.js]: Prefer to avoid CSS selectors in JS - could you add a `js-` prefixed-ID selector to `.utility-search-wrapper`?

File: app/assets/javascripts/fae/navigation/_navigation.js (hunk 1 above)
Comment (jamesmk) [inline app/assets/javascripts/fae/navigation/_navigation.js]: @tshedor this selector was highlighting global search results as well. Let me reconfig it.

**Comment by jamesmk** [comment]
@tshedor Thanks for the feedback, updates made.
